src/Engine/Component/Script.hpp

```diff
@@ -32,18 +32,70 @@ namespace Component {
             
             /// The instance of the script class.
             asIScriptObject* instance = nullptr;
-      
-            /// Entities referenced by this script.
-            std::vector<Entity*> refList;
 
-            ///Map containing the properties, maps a pair of a value and it's type to a name (map<nameOfProperty, pair<typeOfProperty, valueOfProperty>>)
-            std::map<std::string, std::pair<int, void*>> propertyMap;
+            /// Add a property to the propertyMap.
+            /**
+             * @param name The name of the property.
+             * @param type The asTypeID of the property.
+             * @param size The size in number of bytes of the property.
+             * @param data A pointer to the data to store.
+             */
+            ENGINE_API void AddToPropertyMap(std::string name, int type, int size, void* data);
+
+            /// Add a property to the propertyMap.
+            /**
+             * @param name The name of the property.
+             * @param type The asTypeID of the property.
+             * @param target A pointer that points where to copy the data.
+             */
+            ENGINE_API void CopyDataFromPropertyMap(std::string name, void* target);
 
-            /// Fills the property map.
-            void FillPropertyMap();
+            /// Add a property to the propertyMap.
+            /**
+             * @param name The name of the property.
+             * @return A pointer to the data of the property.
+             */
+            ENGINE_API void* GetDataFromPropertyMap(std::string name);
+
+            /// Save the component.
+            /**
+             * @param name The name of the property.
+             * @param type The asTypeID of the property.
+             * @return Is there a property with the provided name and type in propertyMap.
+             */  
+            ENGINE_API bool isInPropertyMap(std::string name, int type);
 
             /// Clears the property map.
             ENGINE_API void ClearPropertyMap();
 
+        private:
+
+            class Property {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') class 'Property' does not have a copy constructor which is recommended since the class contains a pointer to allocated memory. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AnoCopyConstructor)